### PR TITLE
Add function Strings.contains

### PR DESCRIPTION
**Pull request by beutlich**
Cherry-picked from #3247.

### CODE REVIEW

File: Modelica/Utilities/Strings.mo [hunk 1]
```diff
@@ -185,6 +185,54 @@ isEmpty(\"a\");      // returns false
...
+    else
+      result := true;
+    end if;
+
+    annotation (Documentation(info="<html>
+
+<h4>Syntax</h4>
+<blockquote><pre>
+Strings.<strong>contains</strong>(string, searchString);
+Strings.<strong>contains</strong>(string, searchString, caseSensitive=true);
```
Comment (tobolar) [inline Modelica/Utilities/Strings.mo:210]:
```suggestion
result = Strings.<strong>contains</strong>(string, searchString);
result = Strings.<strong>contains</strong>(string, searchString, caseSensitive=true);
```

File: Modelica/Utilities/Strings.mo [hunk 2]
```diff
@@ -185,6 +185,54 @@ isEmpty(\"a\");      // returns false
...
+</pre></blockquote>
+
+<h4>Description</h4>
+<p>
+    Returns true if \"searchString\" is a substring of \"string\". Otherwise, false is returned.
+    The optional argument \"caseSensitive\" controls if substrings match, which differ in case only.
+</p>
+<p>
+    Empty strings are treated as substring of all strings. Therefore, the result is always true if \"searchString\" is empty.
```
Comment (tobolar) [inline Modelica/Utilities/Strings.mo:219]: Why leading spaces here?

File: Modelica/Utilities/Strings.mo [hunk 3]
```diff
@@ -185,6 +185,54 @@ isEmpty(\"a\");      // returns false
...
+<h4>Example</h4>
+<blockquote><pre>
+import Modelica.Utilities.Strings.contains;
+
+contains(\"foobar\", \"OO\");         // returns false
+contains(\"foobar\", \"OO\", false);  // returns true
+contains(\"foo\", \"bar\");           // returns false
+contains(\"foo\", \"\");              // returns false
+contains(\"\", \"\");                 // returns true
+contains(\"foo\", \"\");              // returns true
```
Comment (tobolar) [inline Modelica/Utilities/Strings.mo:231]:
```
contains(\"foo\", \"\");              // returns false/true
```
is doubled here. But with different outputs?

File: Modelica/Utilities/Strings.mo [hunk 4]
```diff
@@ -185,6 +185,53 @@ isEmpty(\"a\");      // returns false
...
+  algorithm
+    if length(searchString) > 0 then
+      result := find(string, searchString, 1, caseSensitive) > 0;
+    else
+      result := true;
+    end if;
+
+    annotation (Documentation(info="<html>
+
+<h4>Syntax</h4>
```
Comment (tobolar) [inline Modelica/Utilities/Strings.mo:207]:
```suggestion
    annotation (Documentation(info="<html>
<h4>Syntax</h4>
```

**Review by tobolar — COMMENTED** [review]
What is the benefit of this function? Just to simplify calling `find` when a boolean output is needed?

**Comment by beutlich** [comment]
> What is the benefit of this function? Just to simplify calling `find` when a boolean output is needed?

Yes, it is a higher-level function. It will be used by #3247 but added here as dedicated issue for better transparency (in release notes etc.)

**Review by tobolar — APPROVED** [review]
Looks good.

**Review by casella — APPROVED** [review]
LGTM

**Pull request merged**